src/test/scala/ir/InterpreterTests.scala

```diff
@@ -1,46 +1,69 @@
 package ir
 
-import analysis.BitVectorEval.*
+import util.PerformanceTimer
+import util.functional._
+import ir.eval._
+import ir.dsl._
 import org.scalatest.funsuite.AnyFunSuite
 import org.scalatest.BeforeAndAfter
 import specification.SpecGlobal
 import translating.BAPToIR
 import util.{LogLevel, Logger}
 import util.IRLoading.{loadBAP, loadReadELF}
-import util.ILLoadingConfig
+import util.{ILLoadingConfig, IRContext, IRLoading, IRTransform}
 
-class InterpreterTests extends AnyFunSuite with BeforeAndAfter {
+def load(s: InterpreterState, global: SpecGlobal): Option[BitVecLiteral] = {
+  val f = NormalInterpreter
 
-  var i: Interpreter = Interpreter()
-  Logger.setLevel(LogLevel.DEBUG)
+  State.evaluate(
+    s,
+    Eval.evalBV(f)(
+      MemoryLoad(SharedMemory("mem", 64, 8), BitVecLiteral(global.address, 64), Endian.LittleEndian, global.size)
+    )
+  ) match {
+    case Right(e) => Some(e)
+    case Left(e) => {
+      None
+    }
+  }
+}
 
-  def getProgram(name: String): (Program, Set[SpecGlobal]) = {
+def mems[E, T <: Effects[T, E]](m: MemoryState): Map[BigInt, BitVecLiteral] = {
+  m.getMem("mem").map((k, v) => k.value -> v)
+}
+
+class InterpreterTests extends AnyFunSuite with BeforeAndAfter {
 
+  Logger.setLevel(LogLevel.WARN)
 
+  def getProgram(name: String): IRContext = {
     val loading = ILLoadingConfig(
       inputFile = s"examples/$name/$name.adt",
       relfFile = s"examples/$name/$name.relf",
```

Comment:
Tests should be updated to use inputs from the tests folder instead of the examples folder (which has mostly been removed)